### ISSUE TITLE
Mistral API is not running while service status reports OK

### ISSUE THREAD

**Issue by armab** [StackStorm/st2-packages#119]
`jessie` failure: https://circleci.com/gh/StackStorm/st2/496

`mistral` service is considered as running by Rspec tests, but in fact it's not fully running, `mistral-api` is not listening on respective port.

```sh
root@7732908eafdc:/# service mistral status
[FAIL] mistral-api is not running ... failed!
[ ok ] mistral-server is running.
root@7732908eafdc:/# echo $?
0
```

`/var/log/mistral-api.log`:
```log
root@7732908eafdc:/var/log/mistral# cat mistral-api.log
[2016-01-27 19:18:29 +0000] [593] [INFO] Starting gunicorn 19.4.5
[2016-01-27 19:18:29 +0000] [593] [INFO] Listening at: http://127.0.0.1:8989 (593)
[2016-01-27 19:18:29 +0000] [593] [INFO] Using worker: sync
[2016-01-27 19:18:29 +0000] [638] [INFO] Booting worker with pid: 638
[2016-01-27 19:18:29 +0000] [639] [INFO] Booting worker with pid: 639
[2016-01-27 19:18:32 +0000] [639] [ERROR] Exception in worker process:
Traceback (most recent call last):
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 515, in spawn_worker
    worker.init_process()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 122, in init_process
    self.load_wsgi()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 130, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/util.py", line 357, in import_app
    __import__(module)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/mistral/api/wsgi.py", line 23, in <module>
    application = app.setup_app()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/mistral/api/app.py", line 49, in setup_app
    db_api_v2.setup_db()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/mistral/db/v2/api.py", line 30, in setup_db
    IMPL.setup_db()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/mistral/db/v2/sqlalchemy/api.py", line 49, in setup_db
    models.Workbook.metadata.create_all(b.get_engine())
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 3687, in create_all
    tables=tables)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1856, in _run_visitor
    conn._run_visitor(visitorcallable, element, **kwargs)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1481, in _run_visitor
    **kwargs).traverse_single(element)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py", line 121, in traverse_single
    return meth(obj, **kw)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 730, in visit_metadata
    _is_metadata_operation=True)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py", line 121, in traverse_single
    return meth(obj, **kw)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 764, in visit_table
    include_foreign_key_constraints=include_foreign_key_constraints
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute
    return meth(self, multiparams, params)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 68, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 968, in _execute_ddl
    compiled
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
    context)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1337, in _handle_dbapi_exception
    util.raise_from_cause(newraise, exc_info)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 199, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
    context)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
    cursor.execute(statement, parameters)
DBError: (psycopg2.ProgrammingError) relation "delayed_calls_v2" already exists
 [SQL: '\nCREATE TABLE delayed_calls_v2 (\n\tcreated_at TIMESTAMP WITHOUT TIME ZONE, \n\tupdated_at TIMESTAMP WITHOUT TIME ZONE, \n\tid VARCHAR(36) NOT NULL, \n\tfactory_method_path VARCHAR(200), \n\ttarget_method_name VARCHAR(80) NOT NULL, \n\tmethod_arguments TEXT, \n\tserializers TEXT, \n\tauth_context TEXT, \n\texecution_time TIMESTAMP WITHOUT TIME ZONE NOT NULL, \n\tprocessing BOOLEAN NOT NULL, \n\tPRIMARY KEY (id)\n)\n\n']
Traceback (most recent call last):
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 515, in spawn_worker
    worker.init_process()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 122, in init_process
    self.load_wsgi()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 130, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/gunicorn/util.py", line 357, in import_app
    __import__(module)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/mistral/api/wsgi.py", line 23, in <module>
    application = app.setup_app()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/mistral/api/app.py", line 49, in setup_app
    db_api_v2.setup_db()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/mistral/db/v2/api.py", line 30, in setup_db
    IMPL.setup_db()
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/mistral/db/v2/sqlalchemy/api.py", line 49, in setup_db
    models.Workbook.metadata.create_all(b.get_engine())
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 3687, in create_all
    tables=tables)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1856, in _run_visitor
    conn._run_visitor(visitorcallable, element, **kwargs)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1481, in _run_visitor
    **kwargs).traverse_single(element)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py", line 121, in traverse_single
    return meth(obj, **kw)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 730, in visit_metadata
    _is_metadata_operation=True)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py", line 121, in traverse_single
    return meth(obj, **kw)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 764, in visit_table
    include_foreign_key_constraints=include_foreign_key_constraints
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute
    return meth(self, multiparams, params)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 68, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 968, in _execute_ddl
    compiled
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
    context)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1337, in _handle_dbapi_exception
    util.raise_from_cause(newraise, exc_info)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 199, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
    context)
  File "/usr/share/python/mistral/local/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
    cursor.execute(statement, parameters)
DBError: (psycopg2.ProgrammingError) relation "delayed_calls_v2" already exists
 [SQL: '\nCREATE TABLE delayed_calls_v2 (\n\tcreated_at TIMESTAMP WITHOUT TIME ZONE, \n\tupdated_at TIMESTAMP WITHOUT TIME ZONE, \n\tid VARCHAR(36) NOT NULL, \n\tfactory_method_path VARCHAR(200), \n\ttarget_method_name VARCHAR(80) NOT NULL, \n\tmethod_arguments TEXT, \n\tserializers TEXT, \n\tauth_context TEXT, \n\texecution_time TIMESTAMP WITHOUT TIME ZONE NOT NULL, \n\tprocessing BOOLEAN NOT NULL, \n\tPRIMARY KEY (id)\n)\n\n']
[2016-01-27 19:18:32 +0000] [639] [INFO] Worker exiting (pid: 639)
[2016-01-27 19:18:33 +0000] [638] [INFO] Worker exiting (pid: 638)
[2016-01-27 19:18:33 +0000] [593] [INFO] Shutting down: Master
[2016-01-27 19:18:33 +0000] [593] [INFO] Reason: Worker failed to boot.
```

**Comment by armab** [comment]
The first thing it's good to fix Rspec/Service to show failure, instead of false-positive "OK" when something like that is happening: 
```sh
service mistral status
[FAIL] mistral-api is not running ... failed!
[ ok ] mistral-server is running.
```

The second is https://github.com/StackStorm/st2-packages/issues/113
Like service should restart all childs if they died.

**Comment by armab** [comment]
@lakshmi-kannan catched this `mistral` error under `el6` too.
See the build: https://circleci.com/gh/StackStorm/st2/501 (`3` tab is `el6` build).

Good to know that for `el6` `service mistral status` shows at least correct "error" exit code:
```sh
# service mistral status
mistral-api is stopped
mistral-server (pid  971) is running...
bash-4.1# echo $?
3
```
So in CircleCI + Rspec tests it shows at least that service is stopped (correct).

For deb/jessie it was `0` OK (incorrect).

**Comment by dennybaa** [comment]
shit still problems. going to debug also.

> Good to know that for el6 service mistral status shows at least correct "error" exit code:

yup. it's the way I made it, mistral - is wrapper (technically not a real service).

**Comment by dennybaa** [comment]
> So in CircleCI + Rspec tests it shows at least that service is stopped (correct).
> For deb/jessie it was 0 OK (incorrect).

Not sure that it's **incorrect**, to say so it's need to look what's status returns for failed service. Just have a look inside init file...

**Comment by dennybaa** [comment]
@armab thanks what's absolutely correct about exit status, I've checked and fixed: https://github.com/dennybaa/st2-packages/commit/514a627184f11c72a5e408993220a4c10b07816e

**Comment by dennybaa** [comment]
merged in #123
